units/patch_intent.go

```diff
@@ -488,40 +486,43 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 	return catcher.Resolve()
 }
 
-func (j *patchIntentProcessor) getPreviousPatchDefinition(project *model.Project, failedOnly bool) ([]patch.VariantTasks, error) {
-	previousPatch, err := patch.FindOne(patch.MostRecentPatchByUserAndProject(j.user.Username(), project.Identifier))
-	if err != nil {
-		return nil, errors.Wrap(err, "error querying for most recent patch")
-	}
-	if previousPatch == nil {
-		return nil, errors.Errorf("no previous patch available")
-	}
-	var res []patch.VariantTasks
-	if failedOnly && !(previousPatch.Status == evergreen.PatchFailed) {
-		return res, nil
-	}
+func setFailedTasksToPrevious(patchDoc, previousPatch *patch.Patch, project *model.Project) error {
```

Comment:
I think this function name isn't quite right. It's really more like setTasksToPreviousFailed or setTopreviousFailedPatchDefinition, right?

Reply:
yes :)